crypto/rand_test.go

```diff
@@ -11,6 +11,14 @@ func TestRandIntnPanics(t *testing.T) {
 			t.Error("expected panic for n <= 0")
 		}
 	}()
-	RandIntn(0)
-	RandIntn(-1)
+
+	_, err := RandIntn(0)
+	if err != nil {
+		t.Error("expected panic on n <= 0, not error")
+	}
+
+	_, err = RandIntn(-1)
+	if err != nil {
+		t.Error("expected panic on n <= 0, not error")
+	}
 }
```

Comment:
This test I believe is wrong, the RandIntn(-1) code is never reached, because the recover triggers instead, which returns the function.

In the recover, you need to call RandIntn(-1) and then you need to above the existing recover set up another recover:

```
defer {
    recover
}
defer {
    recover
    rand -1
}
rand 0
```